### PR TITLE
Bookstore: Redux in React

### PR DESCRIPTION
**Pull request by HammadSiddique**
### In this PR:

- All states are immutable.
- Changed the default state in your books reducer from an empty array to an array with a few books. 
- Imported Redux Provider and my store in the main component of my React app.
- Connected my app to the Redux store.
- The list of books is displayed books from the Redux store

### CODE REVIEW

File: src/components/InputBook.jsx [hunk 1]
```diff
@@ -1,26 +1,50 @@
...
+          className="author"
+          name="author"
+          type="text"
+          placeholder="Author"
+          required
+        />
+        <button type="button" className="add-btn">
+          Add Book
+        </button>
+      </form>
```
Comment (AmaduKamara) [inline src/components/InputBook.jsx]: - [x] Currently, book data from the form is not being added to the boos array in your store. You are correctly fetching the data from the store but are not posting it. You should use the value prop on the input fields and make use of the `onChange` prop then change the being pushed to be the value entered on the input fields. You are required to add a book and that book is displayed on the fly. Please work on that 👍 
- [x] See the Requirement
image


**Hint:**
you could have these properties passed to your input fields and handle the posting
```mdx
value={title}
onChange={(e) => setTitle(e.target.value)}
```
and similar to the author input 👍

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️♻️♻️ 

Hi @HammadSiddique,

AlonzoLeroneThisCloseGIF (3)


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✅ Correct use of Gitflow
✅ No linter error
✅ README is professional
✅ PR is descriptive 👍 

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/components/InputBook.jsx (hunk 1 above)
Comment (HammadSiddique) [reply src/components/InputBook.jsx]: @AmaduKamara thank you for pointing this out, I have added this in my form and now it's working as intended.

**Review by brytebee — APPROVED** [review]
Hi, @HammadSiddique👋,

# STATUS: APPROVED 🕺🥳 

Congratulations🍾🥳!! Your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 



Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**